Toplean_Alexandru/week-02/Domain/Domain/CreateMenuOp/CreateMenuCmd.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Domain.Models;
+
+namespace Domain.Domain.CreateMenuOp
+{
+    public struct CreateMenuCmd
+    {
+        public Restaurant Restaurant { get; }
+        public string Name { get; }
+        public MenuType MenuType { get; }
+
+        public CreateMenuCmd(Restaurant restaurant, string name, MenuType menuType)
+        {
+            Restaurant = restaurant;
+            Name = name;
+            MenuType = menuType;
+        }
+
+        // Returns a tuple. If everything is valid, the function will return true and the MenuErrorCode will be None.
+        // If it is not valid, then the function will return false with the known reason.
+        // Try-Catch block also added in case there is an unknown error and it will return false.
+        public (bool, MenuErrorCode) IsValid()
+        {
+            try
+            {
+                return HasIllegalCharacters(Name) ? (false, MenuErrorCode.IllegalCharacter) :
+                    NameTooShort(Name) ? (false, MenuErrorCode.EmptyField) :
+                    Restaurant == null ? (false, MenuErrorCode.NullRestaurant) :
+                    (true, MenuErrorCode.None);
+            }
+            catch
```

Comment:
When you catch an exception please make sure you record the caught exception by doing ToString on it, this way you will not loose useful information, in this case I would not return just UnknownError because this will not be useful when you have to debug the problem, instead I would prefer that the IsValid method will return the boolean, that tells if the command is valid or not and a string which contains the error message if the input is not valid, and if there was an exception I expect  to have there the exception ToString